include/caffe/net.hpp

```diff
@@ -89,7 +89,7 @@ class Net {
    * @brief For an already initialized net, implicitly copies (i.e., using no
    *        additional memory) the pre-trained layers from another Net.
    */
-  void ShareTrainedLayersWith(Net* other);
+  void ShareTrainedLayersWith(Net* other) const;
```

Comment:
This `const` feels wrong from a logical perspective. It clearly modifies the current network.